packages/salesforcedx-vscode-core/src/conflict/componentDiffer.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { SourceComponent } from '@salesforce/source-deploy-retrieve/lib/src/resolve';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export interface ComponentDiff {
+  projectPath: string;
+  cachePath: string;
+}
+
+export class ComponentDiffer {
+
+  constructor() {}
+
+  public diffComponents(
```

Comment:
let's add some JSDocs on this method too!

Reply:
Sure happy to! Any there any general rules about when to use JSDocs?

Made one for this method in the latest [commit](https://github.com/forcedotcom/salesforcedx-vscode/pull/3332/commits/59311d19c0602fd852eca39f711ae0ccc3b8f0a3).

Reply:
Awesome thanks! We generally try to include JSDocs for methods that have several parameters and cover broader functionality